genie-agent/src/main/java/com/netflix/genie/agent/execution/statemachine/actions/HandleErrorAction.java

```diff
@@ -58,14 +58,14 @@ protected void executePreActionValidation() {
     protected Events executeStateAction(final ExecutionContext executionContext) {
         log.info("Handling execution error");
 
-        final Optional<JobStatus> currentJobStatus = executionContext.getCurrentJobStatus();
+        final Optional<JobStatus> finalJobStatus = executionContext.getFinalJobStatus();
         final Optional<String> claimedJobId = executionContext.getClaimedJobId();
 
-        if (currentJobStatus.isPresent() && claimedJobId.isPresent() && currentJobStatus.get().isActive()) {
+        if (claimedJobId.isPresent() && !finalJobStatus.isPresent()) {
             try {
                 agentJobService.changeJobStatus(
                     claimedJobId.get(),
-                    currentJobStatus.get(),
+                    executionContext.getCurrentJobStatus().get(),
```

Comment:
Might be better here instead of just `get()` to explicitly use `orElse(null)` so it's really clear otherwise I think according to the docs you're going to throw a `NoSuchElementException` [Optional#get](https://docs.oracle.com/javase/8/docs/api/java/util/Optional.html#get--)